pandas/tests/indexes/timedeltas/test_setops.py

```diff
@@ -74,3 +74,66 @@ def test_intersection_bug_1708(self):
         result = index_1 & index_2
         expected = timedelta_range('1 day 01:00:00', periods=3, freq='h')
         tm.assert_index_equal(result, expected)
+
+
+def test_intersection_bug_17391():
```

Comment:
pls just name this test_intersection; this _bug_* scheme is old and not used anymore (you can fix the other one as well). the 'main' intersection tests should be in a test named test_intersection.